### DISCUSSION TITLE
Duvidas em relação ao RunBoat versão 14

### DISCUSSION THREAD

**Discussion by wiprich** [OCA/l10n-brazil#3128] [Q&A]
Ola pessoal, tudo bem?

Estou com o Odoo Enterprise v17 e ta uma relação de amor e ódio com o Odoo.

Eles venderam como se o sistema estivesse "completo", mas vocês sabem melhor do que eu que não esta.

Sou programador, mas não programei nunca em python (PHP, c# AS3, web, etc). Estou enferujado e estou apanhando um pouco com o Odoo.

Atualmente eu tenho uma loja de material elétrico e hidraulico e também uma empresa de energia solar fotovoltaica. A idéia é ter um sistema unificado (Vi no linus tech tips a propagando do odoo, e me interessei e gostei do projeto).

E ai, é aquela coisa. Amor e ódio. Estava em um estágio, que não tinha _**tempo**_ o suficiente para parar e aprender (A loja é nova) . Se o negocio for bom e com custo bom, eu pago. Por isso comprei a versão paga do odoo 17 e esta sendo uma novela. É fantastico o sistema, mas não ta completo, e ta longe de estar completo.

Estou com o PDV rodando, mas não emite cupom, não tem integração com maquininha, não tem nada (por que eles ainda não desenvolveram os modulos). Enfim.

Precisava personalizar o site e colocar um metodo de pagamento com checkout transparente. Mas não tenho acesso ao código para fazer as modificações.

Tentei integrar com o Woocommerce que já roda no nosso site via um plugin pago, mas é meia boca, vai dar pau e gerar mais dor de cabeça. Não quero ter odoo e woocommerce rodando.

Estou a uns 2 meses tentando entender a versão 14 community que me parece ser a mais completa e adaptada ao brasil, mas ainda estou quebrando a cabeça em algumas coisas que pra mim ta dificil (que não é o meu dia a dia, mesmo sendo programador).

Por exemplo. O maldito menu inicial que aperece na versão demo no runboat mas não aparece no baseonly

![image](https://github.com/OCA/l10n-brazil/assets/8609282/7f179685-2580-4f36-bacb-a9fd6f82c460)

No baseonly
![image](https://github.com/OCA/l10n-brazil/assets/8609282/02222477-12a4-475f-8449-def0521036b4)

Como ativa essa "desgraça"... kkkk?

Outra coisa.
Estas são as minhas aplicações do odoo v17 e+

![image](https://github.com/OCA/l10n-brazil/assets/8609282/c8d755dd-5011-4239-a1d9-0b4ef9ddee4a)

A unica coisa que não precisava ali, é a tela da cozinha, o resto, tudo é 10 e vou usar aos poucos.

Alguem me da uma luz?

O que básicamente eu preciso ter pra rodar o sistema com atendimento presencial e vendas web? A partir dai, vou me virando.

Estou com uma instancia na Amazon e vou tentar instalar a versão 14 novamente junto com a localização br.
Existe algum processo automatizado de instalação pra versão br? (vi um script de um gringo que fazia isso)

O trabalho que a comunidade br fez tá 10. Parabens!

**Comment by rvalyi** [comment]
Ola @wiprich , realmente o nivel de désinformaçao sobre o Odoo no Brasil é bem estratosférico desde uns 9 anos... Até 2015, a gente que fez uns 2/3 dessa localização OCA e que co-criou a propria OCA eramos parceiros gold e as coisas eram mais alinhadas com a Odoo SA/Inc. 

So que a gente veio pelo open source no Odoo em 2008/2009. Quando a Odoo SA mudou de licença em 2015 para começar a forçar os parceiros a vender código proprietário (que na epoca era muito carro e zoado), essa "falta de transparência desse novo modelo" não era mais alinhado com o que a gente veio buscar com o Odoo e continuamos as coisas na OCA humildemente fora dos holofotes dos vendedores credenciados da vez...

Nessa caminhada que vc iniciou vc vai descobrir que a questão da instalação/devops não chega a ser 1% do desafio que esta na sua frente. Por isso o foco da OCA não é em pregar uma forma X ou Y de instalar o Odoo e os módulos OCA. Se vc olhar como fazem os 10 principais controbuidores da OCA, vai ver que essas empresas usam 3 ou 4 formas diferentes e que nenhuma delas é algo facil para iniciante... Até as imagens base dos containers Docker a gente nem usa as mesmas. Depende muito do foco: facilidade para descobrir vs escalabilidade vs produtividade em projetos grandes etc... Apenas posso te aconselhar de olhar algumas ferramentas mais basicas que usam os contribuidores aqui no projeto:

- https://github.com/acsone/git-aggregator
https://github.com/acsone/odoo-bedrock
- https://github.com/akretion/docky
- https://github.com/Tecnativa/doodba


Tem coisas mais "prontas" (como instalar os módulos da OCA pelo pip) mas não necessariamente posso aconselhar para depois resolver de vez as questões que vc vai ter na localização.

Bota uma coisa na cabeça: da mesma forma que tem gente que ganhou comissão na suas costas para te vender essa ideia que tava tudo pronto com o Odoo Enterprise v17, entre esse tipo de gente e as pessoas serias que realmente entregam projetos que funcionam no Brasil, que sao dois extremos, existe todo uma variedade de pessoas mais ou menos de bobeira, ou mais ou menos mal intencionadas, para te vender qualquer coisa por qualquer valor que tiver disposto a botar em jogo (e gastar dinheiro com qualquer pessoa tb não é garantia de nada):

![images (2).jpeg](https://github.com/OCA/l10n-brazil/assets/16926/02abadd9-68a1-4e42-9628-753b67546073)

Se vc se perguntar porque o mercado do Odoo esta assim no Brasil, a resposta esta dada aqui por um cara que ganhou o prêmio Nobel da economia 2001 com isso: https://saylordotorg.github.io/text_introduction-to-economic-analysis/s19-01-market-for-lemons.html


Dito isso, se vc realmente quer algo que funciona, te aconselho sondar com as empresas especializadas que fizeram esse projeto aqui que são Akretion, Kmee, Escodoo, Engenere (essas 4 fizeram mais de 99% do código aqui, como pode ser verificado pelos logs e Copyrights nos módulos).

Pode ser que não bate com a ideia que te venderam, mas ai é outra questão... Basicamente o buraco está mais em baixo; tipo muito mais em baixo...

Outro ponto, gostando ou não, a real é que vc ainda esta meses de ter seu Odoo funcionando de vdd para sua empresa (ou ate muitos anos se continuar a fazer as escolhas erradas), nisso eu te aconselho de estudar a versão 14 que hoje é a mais completa mas de cogitar usar a versão 16 em produção pois os módulos importantes que faltam estão chegando la (estamos trabalhando nissi que nem loucos como pode ver se vc segue a atividade do repo).

Bem vindo e boa sorte.

**Reply by wiprich** [reply]
Obrigado pela sua resposta Raphael.

Realmente, uma especialização em uma ferramenta leva muito tempo, seja em uma linguagem ou framework. Eu sou/era dev de games, interatividade (antigo e falecido Flash/as3). Era especialista em as3 e flash na verdade. Enfim. Aplicações comerciais não são meu forte e nem sua lógica. Mas me ajudou bastante. Vou pegar o fio da meada e tocar o barco!

Olhando o repositório da versão 16, percebi que ainda não tem modulo pdv para nfc-e (o que eu preciso). Vou tentar a versão 14, deixar tudo rodando e tentar me envolver na versão 16 depois que eu pegar o jeito da coisa.

Obrigado novamente!

**Reply by rvalyi** [reply]
algumas consideraçoes: pode valer a pena vc fazer alguns apps basicos com Django e com livro de Django para aprender o basico do desenvolvimento web com Python sem ainda encarrar a questao da parte ERP. Agora tb eh importante avaliar bem o folego que tem suas empresas clientes. Veja que eh muito dificil uma empresa de menos de 50 poder contratar alguem os anos que vai precisar para ele aprender a ser um consultor ERP e para nao fazer besteira... E nas empresas maiores, eles podem ate ter o folego so que o desafio para implementar o Odoo ainda eh muito maior, no Brasil em especial... Entao eh bom avaliar bem onde vc esta se metendo antes...

Sou o fundador da empresa Akretion que iniciou o projeto de localizacao brasileira do Odoo em 2009 e fez a maior parte do projeto. Eu e um colega, trabalhamos daqui do Brasil para 2 empresas no ramos dos paneis solares caso tiver interesse. Essa empresa que eh lider da fabricaçao dos paneis hibridos na França: https://dualsun.com e esse cliente deles que eh uma franquiadora da instalacao dos paneis: https://www.soleio.fr 
A gente tem bastante interesse em expandir esses verticais no Brasil caso houvesse gente seria interessada. Sao clientes da gente uns 3 anos ja.

**Comment by antoniospneto** [comment]
O módulo da OCA que ativa esse menu "bonito" é esse aqui:
https://github.com/OCA/web/tree/14.0/web_responsive


Caso queira aprender mais  a implementar e desenvolver no Odoo, antes de tudo recomendo começar com a letirua desse livro:
https://a.co/d/4cSmCVf

**Reply by wiprich** [reply]
Olá Antonio. Obridado pela sua resposta.

Acredite se quiser ou não, esse era o próximo modulo da lista que eu fiz pra comparar as duas bases que eu ia instalar.

Aproveitando o gancho. Na versão comercial eles usam o avalara/avatax para calculo de impostos e ST (coisas que também fogem muito do meu conhecimento).

Percebi que no OCA se usa o IBPT. Ele vai fazer a mesma coisa que o avalara (não necessariamente pois até onde entendi, o avalara é quem gera as notas, etc), ou vai apenas calular os impostos e por fim aplicar aos modulos fiscais. Desculpe se não consegui ser claro.

**Reply by rvalyi** [reply]
na vdd o Avatax deve usar o servidor do impostos do IBPT por tras para uma pequena parte do calculo dos impostos. Essa parte de estimaçao dos impostos do IBPT eh apenas uma parte muito pequena da maluquice fiscal daqui (o que fazemos no modulo l10n_br_fiscal). Cuidado com essa parte ai porque no Brasil eh o lugar onde isso eh o mais complexo do mundo e nao vai aprender apenas em alguns meses. Vc tem que dominar bem essa parte para implementar o Odoo pois nao vai ficar tao mastigado como num ERP proprietario mais maduro pro mercado brasileiro. Eh possivel fazer com o Odoo, mas tem que se garantir muito...

**Reply by antoniospneto** [reply]
@wiprich

Acho que há uma confusão sobre os conceitos de impostos. O IBPT serve apenas para calcular a estimativa dos impostos finais, apresentados apenas nos casos de venda para um consumidor final. Esta é uma informação de referência e não representa a incidência real dos impostos.

Os impostos reais, na localização brasileira da OCA, são calculados internamente, sem uso de APIs ou serviços externos. Atualmente, a localização é bem completa nessa área, atendendo empresas de diferentes regimes, incluindo notas fiscais de produtos e serviços, impostos retidos, substituição tributária, imposto de importação, e muito mais. A localização brasileira está anos-luz à frente do que a Odoo Enterprise oferece em parceria com o AVATAX.

**Comment by wiprich** [comment]
Olá pessoal, tudo bem?

Estou desistindo oficialmente da **versão paga do odoo da própria Odoo SA.** Tem erros toscos de mapeamentos de campos que eles não resolvem. Campos que não fazem parte da localização (avatax americano) estão nas configurações e muitas coisas estão parece que propositalmente lá para confundir e fazer você comprar o "pacote deles", vish, não vou nem entrar no merito. 

É triste ver uma empresa vendendo algo como se fosse completo e pronto para o mercado brasileiro e não esta. Não gera um puto NFC-e. E o modulo da avalara está lá. Prontinho. Ele mesmo acusa que da erro de codigo CSC. Mas não existe nenhum lugar pra colocar o código e integrar. Ai criei um ticket mostrando pra eles as evidencias de que já estava lá a funcionalidade, só precisava configurar. Nem bola deram. Enfim

Desde que iniciei essa discução aprendi muito sobre o Odoo e realmente é um sistema facinante e realmente me apaixonei por ele.  

Mas nao consegui instalar ele **[versão 14 l10n_BR]** com sucesso nas versões mais recentes do ubuntu 24.04. Tentei no 22.04 e no 20.04 e consegui colocar ele para "rodar" sem docker nem nada.

Do nada a amazon interrompeu minha instancia e a imagem caiu fora. Tenho que começar tudo de novo. Estou desapontando [com a amazon] e vou dar mais uma tentiada, mas agora com docker (sou programador do tempo pré-docker, _**menos é mais**_ e muita abstração as vezes só atrapalha (minha visão) ).

[Pergunta]
Existe alguma imagem docker (E aonde encontra-lá, link por gentileza) que já tenha a "mesma" versão que esta no runboat com a configuração de uma empresa do Simples nacional?

Eu preciso do basico, PDV, Vendas e site (Não vou vender em um primeiro momento pelo site, mas integrar com o Wordpress, visto a grande quantidade de plugins e facilidades já existentes).

Eu entrei em contato com algumas empresas mantenedoras do odoo brasileiro. Mas eu entendo, minha empresa não se encaixa em seu publico alvo. Mas fui muito bem atendido por todos que conversei. Mas eu simplesmente não tenho a grana pra pagar a integração.

Se alguem poder me ajudar, fico muito agradecido. Obrigado!

**Reply by rvalyi** [reply]
@wiprich eu te aconselho de ir de v16... Mesmo se vc tiver que quebrar a cabeça para portar um modulo ou outro, no horizonte de 2 anos já sera menos trabalho do que ir de v14, ter que resolver pepinos tb (como https://github.com/OCA/l10n-brazil/issues/2666 que podem ser muito mais difícil resolver na v14 caso te afetar) e ter uma baita dor de cabeça para migrar já que a v14 esta se tornando obsoleto já já. Devemos resolver esse problema em breve https://github.com/OCA/l10n-brazil/issues/3490 essa semana e ai depois ainda tem um bug ou outro, mas o tempo de vc entender como tudo funciona ja deve ficar melhor do que na v14 e no geral o codigo da integração do fiscal como o modulo account no modulo l10n_br_account esta MUITO mais limpo na v16 (porque o modulo account ficou mais limpo). Sobre a Odoo SA, a gente ja sabe que so pode dar nisso infelizmente e a gente ta bem decepcionado dos caras fazer de tudo para fingir que a OCA não existe enquanto os únicos projetos que rodam no Brasil rodam graça a nosso trabalho na OCA. Ainda é uma boa distopia, mas a medida que mais pessoas como vc vão botar a boca no trombone a verdade deve ser restabelecida... Pelo menos isso deve acontecer antes da Odoo conseguir fazer algo minimamente funcional pro Brasil com o Odoo EE...
